python/ray/rllib/agents/dqn/dqn_policy_graph.py

```diff
@@ -380,17 +400,34 @@ def __init__(self, observation_space, action_space, config):
             ("dones", self.done_mask),
             ("weights", self.importance_weights),
         ]
+        print(self.loss_inputs)
+        print(model.state_in)
+        print(model2.state_in)
         TFPolicyGraph.__init__(
             self,
             observation_space,
             action_space,
             self.sess,
-            obs_input=self.cur_observations,
+            obs_input=self.obs_t,
             action_sampler=self.output_actions,
             loss=model.loss() + self.loss.loss,
             loss_inputs=self.loss_inputs,
+            state_inputs=model.state_in + model2.state_in,
+            state_outputs=model.state_out + model2.state_out,
```

Comment:
I think you can pass in the state tensors for model1 to the get_model() for model2. That way, they will share the same set of input placeholders for RNN state.